### PR TITLE
Removed Content property from CSharpStatementIRNode

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - https://github.com/aspnet/Razor/issues/964

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs [hunk 1]
```diff
@@ -61,10 +65,13 @@ public override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIRNode
...
-                var defineSectionStartStatement = new CSharpStatementIRNode()
-                {
-                    Content = /* ORIGINAL: DefineSectionMethodName */ $"DefineSection(\"{sectionName}\", async ({lambdaContent}) => {{",
-                };
+                var defineSectionStartStatement = new CSharpStatementIRNode();
+                RazorIRBuilder.Create(defineSectionStartStatement)
+                    .Add(new RazorIRToken()
+                    {
+                        Kind = RazorIRToken.TokenKind.CSharp,
+                        Content = /* ORIGINAL: DefineSectionMethodName */ $"DefineSection(\"{sectionName}\", async ({lambdaContent}) => {{"
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs]: You can remove `/* ORIGINAL: DefineSectionMethodName */`

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 2]
```diff
@@ -77,20 +77,34 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
...
+                        var padding = BuildOffsetPadding(0, token.Source.Value, Context);
+                        Context.Writer.Write(padding);
+
+                        Context.AddLineMappingFor(token);
+                        Context.Writer.Write(token.Content);
+                    }
+                }
+                else
+                {
+                    Context.Writer.WriteLine(token.Content);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Tokens should not get their own line. We want to be able to generate statements that contain a mix of tokens that have a line mapping and are synthesized.

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 3]
```diff
@@ -77,21 +78,35 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
...
+            var content = builder.ToString();
+
+            IDisposable linePragmaScope = null;
+            if (node.Source != null)
+            {
+                linePragmaScope = Context.Writer.BuildLinePragma(node.Source.Value);
+                var padding = BuildOffsetPadding(0, node.Source.Value, Context);
+                Context.Writer
+                    .Write(padding)
+                    .Write(content);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Do a series of writes here with a loop. Make sure it ends with a newline

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 4]
```diff
@@ -77,21 +78,31 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
...
-                    Context.Writer.Write(node.Content);
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                if (!(node.Children[i] is RazorIRToken token) || !token.IsCSharp)
+                {
+                    continue;
                 }
+
+                Context.AddLineMappingFor(token);
+                Context.Writer.Write(token.Content);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Take a look at https://github.com/aspnet/Razor/pull/1115/files#diff-8df250e114e6022d67d79cdd0e5c4b56L39

This handles extension nodes as well.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Comment by ajaybhargavb** [comment]
Looks like this needs a reaction in Mvc because Mvc isn't using `Razor.Extensions` yet.

**Comment by rynowak** [comment]
You can also just wait, because that changeover should happen Monday.

**Comment by ajaybhargavb** [comment]
Rebased with dev

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 5]
```diff
@@ -77,31 +78,48 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
+            var isWhitespaceToken = node.Children.Count == 1 &&
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Could just do a `node.Children.All` here

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs [hunk 6]
```diff
@@ -209,25 +210,42 @@ public override void VisitCSharpAttributeValue(CSharpAttributeValueIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
-            if (string.IsNullOrWhiteSpace(node.Content))
+            if (node.Children.Count == 1)
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs]: Same comment about `.All` here

**Pull request merged**